src/oops/SOLID/openClosePrinciple/before/taxes/TaxCalculatorFactory.java

```diff
@@ -0,0 +1,22 @@
+package oops.SOLID.openClosePrinciple.before.taxes;
+
+import oops.SOLID.openClosePrinciple.before.employees.Employee;
+import oops.SOLID.openClosePrinciple.before.employees.FullTimeEmployee;
+import oops.SOLID.openClosePrinciple.before.employees.Intern;
+import oops.SOLID.openClosePrinciple.before.employees.PartTimeEmployee;
+
+import java.security.InvalidParameterException;
+
+public class TaxCalculatorFactory {
+    public static Taxable getTaxCalculator(Employee employee) {
+        if (employee instanceof FullTimeEmployee) {
+            return new FullTimeEmployeeTaxCalculator();
+        } else if (employee instanceof PartTimeEmployee) {
+            return new PartTimeEmployeeTaxCalculator();
+        } else if (employee instanceof Intern) {
+            return new InternTaxCalculator();
+        } else {
+            throw new InvalidParameterException("Employee Type not supported.");
```

Comment:
Should we use this exception, or UnsupportedOperationException? In general, how do you decide which exception to use?